Modernize avoid a C array

r-barnes opened the pull request:
Differential Revision: D29717118

malfet commented on torch/lib/libshm/core.cpp:
```diff
@@ -46,14 +46,13 @@ void start_manager() {
   for (;;) {
-    SYSCHECK_ERR_RETURN_NEG1(bytes_read = read(pipe_ends[0], buffer, sizeof(buffer)));
-    handle.append(buffer, bytes_read);
+    SYSCHECK_ERR_RETURN_NEG1(bytes_read = read(pipe_ends[0], buffer.data(), buffer.size()));
```
```suggestion
    auto bytes_read = read(pipe_ends[0], buffer.data(), buffer.size());
    SYSCHECK_ERR_RETURN_NEG1(bytes_read);
```

malfet commented on torch/lib/libshm/core.cpp:
```diff
@@ -46,14 +46,13 @@ void start_manager() {
-    handle.append(buffer, bytes_read);
+    SYSCHECK_ERR_RETURN_NEG1(bytes_read = read(pipe_ends[0], buffer.data(), buffer.size()));
+    handle.append(buffer.data(), bytes_read);
     if (bytes_read == 0 || handle[handle.length() - 1] == '\n') {
```
Nit (and imo, it would be better to replace `for(;;)` with `while(handle.empty() || handle.back() != '\n')`
```suggestion
    if (bytes_read == 0 || handle.back() == '\n') {
```

r-barnes committed:
Modernize avoid a C array (#61740)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/61740

Test Plan: Sandcastle

Reviewed By: malfet

Differential Revision: D29717118

fbshipit-source-id: 87d1238ec02518892dac877b597e88a5fe352525
